examples/bls.rs

```diff
@@ -0,0 +1,29 @@
+    extern crate rand;
+    extern crate bn;
+    extern crate ring;
+    #[macro_use]
+    extern crate arrayref;
+
+
+    use bn::{Group,Fr, G1, G2, pairing};
+    use ring::digest;
+
+    fn main() {
+        let rng = &mut rand::thread_rng();
+
+        // Generate Keys
+        let alice_sk = Fr::random(rng);
+
+        // Generate Public Keys
+        let alice_pk = G2::one() * alice_sk;
+        // Generate Signature
+        let mut ctx = digest::Context::new(&digest::SHA512);
+        ctx.update(b"Hello!");
+        let hash = ctx.finish();
+        let msgm1 =Fr::interpret(array_ref![hash.as_ref(),0,64]);
+        let sigm1_a = msgm1 * alice_sk;
+
+        // Verify single signature
+        assert!(pairing(G1::one() * sigm1_a, G2::one())== pairing(G1::one() * msgm1, alice_pk));
```

Comment:
Assuming the README is correct that `*` is scalar multiplication, and not a seemingly unnecessary group addition, then..

@Liamsi is correct.  You cannot "hash to the curve" with a scalar multiplication like this.  And this is not verification operation from the BLS signature scheme.  

Anyone who sees the message knows `msgm1`.  If they see `sigm1_a` too, then can compute `alice_sk` by simple division mod the order of the scalars.  If you only give them `G1::one() * sigm1_a` then they can still forge signatures because `G1::one() * sigm2_a = (G1::one() * sigm1_a) * (msgm2 / msgm1)`.